tests/gtest/fake_platform.cpp

```diff
@@ -139,24 +139,39 @@ void FakePlatform::ProcessAlarm(uint64_t &aTimeout)
     {
         alarm = &mMilliAlarmStart;
     }
+    if (mReceiveAtStart < *alarm)
+    {
+        alarm = &mReceiveAtStart;
+    }
+    else if (mReceiveAtEnd < *alarm)
+    {
+        alarm = &mReceiveAtEnd;
+    }
 
     if (mNow < *alarm)
     {
         aTimeout -= *alarm - mNow;
         mNow = *alarm;
     }
     *alarm = kAlarmStop;
-#if OPENTHREAD_CONFIG_PLATFORM_USEC_TIMER_ENABLE
-    if (alarm == &mMicroAlarmStart)
+    if (alarm == &mReceiveAtStart)
     {
-        otPlatAlarmMicroFired(mInstance);
+        mChannel = mReceiveAtChannel;
     }
-    else
-#endif
-        if (alarm == &mMilliAlarmStart)
+    if (alarm == &mReceiveAtEnd)
+    {
+        mChannel = 0;
+    }
+    else if (alarm == &mMilliAlarmStart)
```

Comment:
I did some refactor in this PR for the concern about `ProcessAlarm`.

For `mMilliAlarmStart` happens to be the same as `mReceiveAtEnd`. It should be fine, because this method triggers one of the earliest schedules and will trigger others in next calls. PTAL.